### PR TITLE
Fix: Make the developer guide less outdated

### PR DESCRIPTION
**Pull request by onebeastchris**

### CODE REVIEW

File: _docs/other/developer-guide.md [hunk 1]
```diff
@@ -3,14 +3,18 @@ title: Developer Guide
 ---
 
 ## Compiling
-1. Clone the repo to your computer (EG: `git clone https://github.com/GeyserMC/Geyser.git`)
-2. [Install Maven](https://maven.apache.org/install.html)
-3. Navigate to the Geyser root directory and run `git submodule update --init --recursive`. This downloads all the needed submodules for Geyser and is a crucial step in this process.
-4. Run `mvn clean install` and locate to the `bootstrap`, then your desired Geyser version, then `target` folder.
+1. Clone the repo to your computer.
+2. Navigate to the Geyser root directory and run `git submodule update --init --recursive`. This command downloads all the needed submodules for Geyser and is a crucial step in this process.
+3. Run `gradlew build` and locate to `bootstrap/<platform>/build/` folder.
```
Comment (Chew) [inline _docs/other/developer-guide.md:8]:
```suggestion
3. Run `./gradlew build` and locate to `bootstrap/<platform>/build/` folder.
```

File: _docs/other/developer-guide.md [hunk 2]
```diff
@@ -3,14 +3,18 @@ title: Developer Guide
...
 
 ## Project layout
+
+Geyser's code is divided into different modules. For example:
+
 * `bootstrap` is where we hold the specific platform code. So if you're porting Geyser to a new platform, or working with platform-specific code, you likely want to be in here.
-* `core` is where connections are handled and the data conversion is done. The majority of Geyser work is held here.
+* `core` is where connections are handled and the data/packet conversion is done. The majority of Geyser work is held here.
+* `api` is where the Geyser API is found.
+* `build-logic` contains the kotlin gradle plugins that are used to build Geyser.
```
Comment (Chew) [inline _docs/other/developer-guide.md:17]: The wording is inconsistent here, "is where" to "contains"

**Review by Chew — COMMENTED** [review]
Some tiny comments :)

File: _docs/other/developer-guide.md (hunk 2 above)
Comment (onebeastchris) [reply _docs/other/developer-guide.md:17]: Thought I'd mix it up to make it less repetitive, but sure, I'll revert

Comment (Chew) [reply _docs/other/developer-guide.md:17]: Either works, this isn't really an academic essay that would require breaking up the flow tho 😄

**Comment by Camotoy** [comment]
We should document the MCDev IntelliJ plugin, since that's what I use to browse NBT. I'm not sure if there's something similar for VS Code.

File: _docs/other/developer-guide.md [hunk 3]
```diff
@@ -3,56 +3,71 @@ title: Developer Guide
...
+
+### Lombok
 
 *Note that the latest version of IntelliJ IDEA does not require additional setup in order to develop with Geyser.*
 
-If you're using an IDE for editing any of the GeyserMC projects you will most likely need to install the Project Lombok plugin as it is used to generate a bunch of handy functions. You can edit without it but you may get missing functions and or other issues displayed in your IDE. Please see the IDE section on their site for the supported plugins and install methods [https://projectlombok.org/setup/overview](https://projectlombok.org/setup/overview).
+If you're using an IDE for editing any of the GeyserMC projects you will most likely need to install the Project Lombok plugin as it is used to generate a bunch of handy functions. 
+You can edit without it but you may get missing functions and or other issues displayed in your IDE. Please see the IDE section on their site for the supported plugins and install methods [https://projectlombok.org/setup/overview](https://projectlombok.org/setup/overview).
+
+###
```
Comment (Chew) [inline _docs/other/developer-guide.md:35]: What is this random header?

File: _docs/other/developer-guide.md [hunk 4]
```diff
@@ -3,56 +3,71 @@ title: Developer Guide
...
-## GopherTunnel
+### GopherTunnel
 [GopherTunnel](https://github.com/Sandertv/gophertunnel/tree/master/minecraft/protocol/packet) is a library for Bedrock Edition written in Go. The source code is an excellent documentation for the Bedrock protocol.
 
-## wiki.vg
+### wiki.vg
 For a full rundown of the Java Edition protocol, see [here](https://wiki.vg/Protocol).
 
-## wiki.vg (Bedrock)
+### wiki.vg (Bedrock)
```
Comment (Chew) [inline _docs/other/developer-guide.md:45]: Does this need a separate header?

**Review by Chew — COMMENTED** [review]
I just keep nitpicking I'm so sorry :(

File: _docs/other/developer-guide.md (hunk 4 above)
Comment (onebeastchris) [reply _docs/other/developer-guide.md:45]: All programs/pages have a header at the moment, just kept it - added another `#` as that's less "blown up"
Current page:
image
Latest changes:
image

Comment (Chew) [reply _docs/other/developer-guide.md:43]: I meant like, wiki.vg is one program with distinct links. Can't it just be like

**Java** - blah<br>
**Bedrock** - blah

Comment (onebeastchris) [reply _docs/other/developer-guide.md:43]: sure; but that's been there before.. I'm not really looking to rework the development guide here, that's for another PR

Comment (Chew) [reply _docs/other/developer-guide.md:43]: Fair enough, can handle that later :)

**Review by Chew — APPROVED** [review]

**Pull request merged**